common/include/pcl/experimental/executor/default/cuda_executor.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#ifdef CUDA
+  #include <cuda_runtime_api.h>
+#endif
+
+#include <pcl/experimental/executor/default/base_executor.hpp>
+#include <pcl/experimental/executor/default/inline_executor.hpp>
+
+namespace executor {
+
+#ifdef CUDA
+template <typename F>
+__global__ void global_kernel(F f) {
+  f();
+}
+#endif
+
+template <typename Blocking, typename ProtoAllocator>
+struct cuda_executor;
+
+#ifdef CUDA
+template <>
+struct is_executor_available<cuda_executor> : std::true_type {};
+#endif
+
+template <typename Blocking = blocking_t::always_t,
+          typename ProtoAllocator = std::allocator<void>>
+struct cuda_executor {
+  using shape_type = typename std::array<int, 6>;
```

Comment:
`dim3` is an alias of `uint3`. You should store here an unsigned type as well.

Note to self: expect to see a split into grid and block shape in future commits.